aya/src/programs/links.rs

```diff
@@ -257,4 +297,32 @@ mod tests {
         assert!(*l1_detached.borrow() == 1);
         assert!(*l2_detached.borrow() == 1);
     }
+
+    #[test]
+    fn test_forget() {
```

Comment:
this should be `test_owned_detach` and we need a  new`test_owned_drop` that doesn't call `detach()` but checks that the link gets detached on drop